### PR TITLE
Update App Notifications

### LINKED ISSUE #12547: Spending limit notification is too sticky
Following up from #12467

The observed behavior now (at least in the preview environment I tested) is that once the spending limit notification become visible, it does not go away until dismissed (by clicking on the `X`). This happens even if the user navigates to another page, or the page is refreshed, or the user logs out and in again.

This is problematic especially after user has managed to get the spending limit increased or changes to a different billing account whose spending limit has not been reached yet.

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description
Dismiss Usage Limit notifications automatically on resolution, e.g. on increase of Usage Limit.

## Related Issue(s)

Fixes #12547

## How to test
* Have 2 accounts on a team with UB enabled.
* Decrease usage limit and drive into the limit notification. 
* Increase the spending limit on one account.
* See the notification is dismissed for both accounts.

## Release Notes

```release-note
Dismiss Usage Limit notifications automatically.
```

## Werft options:

- [x] /werft with-preview
- [x] /werft with-payment

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by AlexTugarev** [comment]
@jldec, please have a look. 🙏🏻 
(Hopefully, the latest build will succeed 🤞🏻 )

**Comment by svenefftinge** [comment]
tbh, it seems a bit overkill to roll out push events for this. @jldec why do we need to show the limit everywhere in the dashboard? Would it not be sufficient to check and display the limit on workspace-start?

**Comment by jldec** [comment]
@svenefftinge, @AlexTugarev, the idea behind a global dashboard notification from the start was that this message is really intended for team owners who may not be starting workspaces all the time. (We also envisaged email notifications at some (80%) threshold before the limit is reached.)

If the notification mechanism is _really_ expensive (implementation cost, complexity), we could pull back and ship just the notification/error on start workspace for now, but if we have something that works globally, I believe users would prefer that because it makes the out-of-credits status much clearer.

**Comment by svenefftinge** [comment]
> this message is really intended for team owners who may not be starting workspaces all the time

I don't think people who don't start workspaces hang out on the dashboard. They would get nudged by developers who see the information when starting a workspace.

**Comment by jldec** [comment]
@svenefftinge Maybe I'm nitpicking, but the UX of a dashboard notification does feel superior. Happy to roll-back if you think the implementation or complexity cost is not worth it.

#### Error-on-start only (no dashboard notification):
- developer gets error on workspace start
- contacts team owner
- team owner first confirms error by starting a workspace
- team owner sees error and confirms that it is because "usage limit exceeded"
 
#### Dashboard notification
- any team member sees error (on workspace start or in dashboard)
- contacts team owner
- team owner immediately sees "usage limit exceeded" in dashhboard - no additional confirmation required

**Review by jldec — APPROVED** [review]
Thank you @AlexTugarev
LGTM - tested as described in PR description and works as advertise 

/hold in case you decide to pull back, based on discussion with @svenefftinge.

**Comment by svenefftinge** [comment]
Anyone should see the status of usage vs limits when looking at the billing page - no need to start a workspace. But I shared my concerns and don't want to further block this if you think it's good.

File: components/server/ee/src/workspace/gitpod-server-impl.ts
Comment (geropl) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:2188]: nit: feels like a debugging leftover.

File: components/server/src/messaging/local-message-broker.ts
Comment (geropl) [inline components/server/src/messaging/local-message-broker.ts:173]: copy'n'paste, should read "listenToSubscriptionUpdates"

File: components/dashboard/src/AppNotifications.tsx
Comment (geropl) [inline components/dashboard/src/AppNotifications.tsx:25]: :cloud: Feels a bit off that we have the "notifications" abstraction but need to know about "subscription updates" and understand them to know when to reload notifications.

:bulb: Did you think about pushing notifications instead? Either the messages, or alternatively just the signal to reload (might be the cleaner solution, actually). That way, `subscriptionUpdate` could stay backend only.

**Review by geropl — APPROVED** [review]
> Anyone should see the status of usage vs limits when looking at the billing page - no need to start a workspace. But I shared my concerns and don't want to further block this if you think it's good.

I feel having (reactive) notifications like this can lift the UX to a new level, and I think we should aim for that. Notifications like this have been requested for a couple of usecases and the abstraction and mechanism solves the it nicely. Treating the paywall as a first "test driver" felt like the natural choice.

@AlexTugarev I left some cleanup comments, and [a potential follow-up](https://github.com/gitpod-io/gitpod/pull/12670#discussion_r970947043), but besides: Looks good to go! :+1:

File: components/server/src/messaging/local-message-broker.ts
Comment (AlexTugarev) [reply components/server/src/messaging/local-message-broker.ts]: done

File: components/server/ee/src/user/eligibility-service.ts
Comment (AlexTugarev) [reply components/server/ee/src/user/eligibility-service.ts]: done

File: components/server/ee/src/workspace/gitpod-server-impl.ts
Comment (AlexTugarev) [reply components/server/ee/src/workspace/gitpod-server-impl.ts]: done

**Comment by AlexTugarev** [comment]
Thanks for the review, @geropl! Changes applied. Build is running...

**Comment by AlexTugarev** [comment]
Fixed compile errors after rebase merge conflicts.

**Comment by AlexTugarev** [comment]
```
Typechecking Werft Typescript files
STDOUT: platform-delete-preview-environment.ts(3,10): error TS2305: Module '"./util/preview"' has no exported member 'HarvesterPreviewEnvironment'.
platform-delete-preview-environment.ts(3,39): error TS2305: Module '"./util/preview"' has no exported member 'PreviewEnvironment'.
```
looks unrelated. updated again.

**Pull request merged**
